src/machines/java/com/enderio/machines/common/blockentity/ZombieGeneratorBlockEntity.java

```diff
@@ -0,0 +1,146 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.api.capacitor.CapacitorModifier;
+import com.enderio.api.capacitor.QuadraticScalable;
+import com.enderio.core.common.sync.FluidStackDataSlot;
+import com.enderio.core.common.sync.SyncMode;
+import com.enderio.machines.common.blockentity.base.PowerGeneratingMachineEntity;
+import com.enderio.machines.common.io.fluid.MachineFluidHandler;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import com.enderio.machines.common.menu.ZombieGeneratorMenu;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+import org.jetbrains.annotations.Nullable;
+
+import static com.enderio.base.common.init.EIOFluids.DEW_OF_THE_VOID;
+import static com.enderio.base.common.init.EIOFluids.NUTRIENT_DISTILLATION;
+
+public class ZombieGeneratorBlockEntity extends PowerGeneratingMachineEntity {
```

Comment:
Maybe rename this to something along these lines to make slightly more sense? And also for clarity, make this an abstract class and the default behaviour another subclass :)

```suggestion
public class MobGeneratorBlockEntity extends PowerGeneratingMachineEntity {
```